include/deal.II/dofs/dof_tools.h

```diff
@@ -1153,19 +1153,20 @@ namespace DoFTools
    *
    * @author Matthias Maier, 2012 - 2015
    */
-  template <typename FaceIterator>
+  template <typename FaceIterator, typename number>
   void
   make_periodicity_constraints(
     const FaceIterator &                         face_1,
     const typename identity<FaceIterator>::type &face_2,
-    AffineConstraints<double> &                  constraints,
+    AffineConstraints<number> &                  constraints,
     const ComponentMask &            component_mask   = ComponentMask(),
     const bool                       face_orientation = true,
     const bool                       face_flip        = false,
     const bool                       face_rotation    = false,
     const FullMatrix<double> &       matrix           = FullMatrix<double>(),
     const std::vector<unsigned int> &first_vector_components =
-      std::vector<unsigned int>());
+      std::vector<unsigned int>(),
+    const number periodicity_factor = 1.);
```

Comment:
I like this interface. I think I always thought about applying a phase shift in a too complicated manner (complex valued phase-shift rotation matrix instead of splitting the complex phase shift and the matrix-valued rotation). This seems to be a very sensible and pragmatic alternative.